usr.sbin/bhyve/migration.c

```diff
@@ -761,155 +653,95 @@ find_entry_for_dev(const char *name)
-	msg.len = data_size;
-	strncpy(msg.name, dev, MAX_DEV_NAME_LEN);
+		rc = (*dev_info->snapshot_cb)(meta);
+		if (rc < 0) {
+			DPRINTF("Could not get info about %s dev", dev);
+			return (-1);
+		}
 
-	rc = migration_send_data_remote(socket, &msg, sizeof(msg));
-	if (rc < 0) {
-		fprintf(stderr,
-			"%s: Could not send msg for %s dev\r\n",
-			__func__,
-			dev);
-		return (-1);
-	}
+		data_size = vm_get_snapshot_size(meta);
 
-	if (data_size == 0) {
-		fprintf(stderr, "%s: Did not send %s dev. Assuming unused. "
-			"Continuing...\r\n", __func__, dev);
-		return (0);
+		msg.type = MESSAGE_TYPE_DEV;
+		msg.len = data_size;
+		strlcpy(msg.name, dev, MAX_DEV_NAME_LEN);
 	}
 
-	rc = migration_send_data_remote(socket, buffer, data_size);
+	rc = migration_transfer_data(socket, &msg, sizeof(msg), req);
 	if (rc < 0) {
-		fprintf(stderr,
-			"%s: Could not send %s dev\r\n",
-			__func__,
-			dev);
+		DPRINTF("Could not transfer msg for %s dev", dev);
 		return (-1);
 	}
 
-	return (0);
-}
-
-static int
-migrate_recv_dev(struct vmctx *ctx, int socket, char *buffer, size_t len)
-{
-	int rc;
-	size_t data_size;
-	struct migration_message_type msg;
-	struct vm_snapshot_meta *meta;
-	const struct vm_snapshot_dev_info *dev_info;
-
-	memset(&msg, 0, sizeof(msg));
+	if (req == MIGRATION_RECV_REQ) {
+		if (msg.type != MESSAGE_TYPE_DEV) {
+			DPRINTF("Wrong message type for device.");
+			return (-1);
+		}
 
-	rc = migration_recv_data_from_remote(socket, &msg, sizeof(msg));
-	if (rc < 0) {
-		fprintf(stderr, "%s: Could not recv msg for device.\r\n", __func__);
-		return (-1);
+		data_size = msg.len;
 	}
 
-	data_size = msg.len;
+	if (data_size == 0)
+		return (0); // this type of device is not used
 
-	if(data_size == 0) {
-		fprintf(stderr, "%s: Did not restore %s dev. Assuming unused. "
-			"Continuing...\r\n", __func__, msg.name);
-		return (0);
-	}
 
-	memset(buffer, 0 , len);
-	rc = migration_recv_data_from_remote(socket, buffer, data_size);
+	rc = migration_transfer_data(socket, buffer, data_size, req);
 	if (rc < 0) {
-		fprintf(stderr,
-			"%s: Could not recv %s dev\r\n",
-			__func__,
-			msg.name);
+		DPRINTF("Could not transfer %s dev", dev);
 		return (-1);
 	}
 
-	dev_info = find_entry_for_dev(msg.name);
-	if (dev_info == NULL) {
-	    fprintf(stderr, "%s: Could not find the device %s "
-		    "or migration not implemented yet for it."
-		    "Please check if you have the same OS version installed.\r\n",
-		    __func__, msg.name);
-	    return (0);
-	}
-
-	meta = &(struct vm_snapshot_meta) {
-		.ctx = ctx,
-		.dev_name = msg.name,
-
-		.buffer.buf_start = buffer,
-		.buffer.buf_size = data_size,
-
-		.op = VM_SNAPSHOT_RESTORE,
-	};
-
-	meta->buffer.buf = meta->buffer.buf_start;
-	meta->buffer.buf_rem = meta->buffer.buf_size;
+	if (req == MIGRATION_RECV_REQ) {
+		dev_info = find_entry_for_dev(msg.name);
+		if (dev_info == NULL) {
+			DPRINTF("Could not find the device %s "
+				"or migration not implemented yet for it.", msg.name);
```

Comment:
This is done using `DPRINTF`, while the other similar print is done using `EPRINTF`.